includes/polling/os/zxdsl.inc.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+use LibreNMS\RRD\RrdDefinition;
+
+if (strstr($device['sysObjectID'], '.1.3.6.1.4.1.3902.')) {
+    $sysObjectId = str_replace('.1.3.6.1.4.1.3902.', '', $device['sysObjectID']);
+    $sysObjectIdSplit = explode('.', $sysObjectId);
+    if (count($sysObjectIdSplit) >= 2) {
+        $hardware = "ZXDSL ".$sysObjectIdSplit[1];
+    }
```

Comment:
This is a way to get the model if (and only if) ZTE is consistant in the ObjectID and model numbers. But I would be surprised they don't have a dedicated OID that provides the hardware model and variant. You can snmpwalk the device completely and search for the model name in the resulting file.

Reply:
Sure, this depends on the consistancy between ObjectID and Model, but there is no other place in the whole snmpwalk containing the model (ZXDSL9816 and 9806, have no others to check)
I will address this question to ZTE if there is another way to get the model.

Reply:
They told me that the ObjectID is the correct way to find the model number.
Also there is an additional character in the ObjectID.
First example the ObjectID ends with "9806.22.1.1.1"
9806 -> Model
22 -> Additional character, 22. character of the latin alphabet
1.1.1 -> End of the Model
The model here is ZXDSL 9806V

Second example, the ObjectID ends with "9816.1.1.1"
9806 -> Model
1.1.1 -> End of the Model
The model here is ZXDSL 9816

I will add this in my next commit